### PR TITLE
Upgrade android libraries versions

### PR DESCRIPTION
**Pull request by GabrielFleischer**
While writing some tests, I added a library that needed SDK 31.

So I upgraded it and took time to upgrade other libraries in the meantime.

### CODE REVIEW

File: fe2-android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -16,7 +16,9 @@
     android:roundIcon="@mipmap/ic_launcher_round"
     android:supportsRtl="true"
     android:theme="@style/AppTheme">
-    <activity android:name="com.github.dedis.popstellar.ui.home.HomeActivity">
+    <activity
+      android:name="com.github.dedis.popstellar.ui.home.HomeActivity"
+      android:exported="false">
```
Comment (arnauds5) [inline fe2-android/app/src/main/AndroidManifest.xml:21]: With exported="false", the app won't launch and the IDE says "A launchable activity must be exported as of Android 12"
```suggestion
      android:exported="true">
```

**Review by arnauds5 — CHANGES_REQUESTED** [review]

File: fe2-android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/main/AndroidManifest.xml]: This is ... weird ... I don't have this issue and from the documentation, and exported activity can be launched from other apps.

I will fix this

Comment (arnauds5) [reply fe2-android/app/src/main/AndroidManifest.xml]: I was using the emulator with Pixel 2, API 29.
Do we need to update something in android studio ?

**Review by arnauds5 — APPROVED** [review]

**Review by jbsv — APPROVED** [review]
LGTM

**Pull request merged**